django/db/backends/postgresql/schema.py

```diff
@@ -104,6 +104,15 @@ def _alter_column_type_sql(self, table, old_field, new_field, new_type):
 
     def _alter_field(self, model, old_field, new_field, old_type, new_type,
                      old_db_params, new_db_params, strict=False):
+        # Drop indexes on varchar/text columns that are changing to a different
+        # type.
+        if old_field.db_index and (
```

Comment:
Should this be `old_field.db_index or old_field.unique` since `unique=True` also creates indexes? (at least until https://code.djangoproject.com/ticket/24082 is fixed) (I made this comment previously but it's now hidden and I don't think you responded.)